dace/transformation/transformation.py

```diff
@@ -1,67 +1,55 @@
-# Copyright 2019-2021 ETH Zurich and the DaCe authors. All rights reserved.
+# Copyright 2019-2022 ETH Zurich and the DaCe authors. All rights reserved.
 """
-Contains classes that represent data-centric transformations.
-
-There are three general types of transformations:
-  * Pattern-matching Transformations (extending Transformation): Transformations
-    that require a certain subgraph structure to match.
-  * Subgraph Transformations (extending SubgraphTransformation): Transformations
-    that can operate on arbitrary subgraphs.
-  * Library node expansions (extending ExpandTransformation): An internal class
-    used for tracking how library nodes were expanded.
+This file contains classes that describe data-centric transformations.
+
+All transformations extend the ``TransformationBase`` class. There are three built-in types of transformations in DaCe:
+  * Pattern-matching Transformations (extending ``PatternTransformation``): Transformations
+    that require a certain subgraph structure to match. Within this abstract class, there are two sub-classes:
+    * ``SingleStateTransformation``: Patterns are limited to a single SDFG state.
+    * ``MultiStateTransformation``: Patterns are given on a subgraph of an SDFG state machine.
+    A pattern-matching must extend at least one of those two classes.
+  * Subgraph Transformations (extending SubgraphTransformation): Transformations that can operate on arbitrary
+    subgraphs.
+  * Library node expansions (extending ExpandTransformation): An internal class used for tracking how library nodes
+    are expanded.
+
+Some transformations are included in the SDFG Dataflow Coarsening pass. In order to declare a transformation as part
+of the dataflow coarsening pass, it should also extend the ``DataflowCoarseningTransformation`` mixin.
```

Comment:
With such names, we will soon need another increase in line width :p

Reply:
We have another PR replacing it with `simplify` :)